heat/core/logical.py

```diff
@@ -519,6 +519,9 @@ def sanitize_input_type(
     # if one of the tensors is distributed, unsplit/gather it
     if x.split is not None and y.split is None:
         t1 = manipulations.resplit(x, axis=None)
+        local_start = x.comm.chunk(t1.gshape, x.split)[0]
```

Comment:
x.comm.chunk() gives the offsets for a balanced DNDarray on all processes, but this is not always the case. Calculating the offset from x.lshape_map may give more accurate values of the current distribution.